src/averageAges.js

```diff
@@ -56,6 +71,16 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const child = (person) => (people.some(one => one.name === person.mother));
+  const son = (person) => (child(person) && person.sex === 'm');
+  const sons = people.filter(onlyWithSon ? son : child);
+  const mothers
+    = people.filter(each => sons.some(one => one.mother === each.name));
```

Comment:
check all similar cases

```suggestion
  const mothers = people.filter(person => (
    sons.some(son => son.mother === person.name)
  );
```